ion-c-sys/src/reader.rs

```diff
@@ -347,7 +347,32 @@ impl<'a> IonCReaderHandle<'a> {
         Ok(value)
     }
 
-    // TODO ion-rust/#42 - support decimal reads
+    /// Reads a `bigdecimal` value from the reader.
+    ///
+    /// ## Usage
+    /// ```
+    /// # use std::convert::*;
+    /// # use bigdecimal::BigDecimal;
+    /// # use ion_c_sys::*;
+    /// # use ion_c_sys::decimal::*;
+    /// # use ion_c_sys::reader::*;
+    /// # use ion_c_sys::result::*;
+    /// # fn main() -> IonCResult<()> {
+    /// let mut reader = IonCReaderHandle::try_from("3.0")?;
```

Comment:
I don't know if there's a community or `rustfmt`-driven recommendation here, but for the humans among us I think indenting the contents of this function would be helpful. :)

Reply:
I pulled this style from one of "the books" ;-)--the example docs don't get the `main` scaffolding, so the block isn't there in the code.

If I do indent the code, I believe it will have an extra level of indentation in the rendered docs...

Reply:
The style is consistent with:

https://doc.rust-lang.org/rustdoc/documentation-tests.html

I'm not a fan, to be honest, but it will make our docs look a bit uglier to indent.

If you like I can cut an issue to address this globally.

Reply:
> If I do indent the code, I believe it will have an extra level of indentation in the rendered docs...

I didn't realize this, thanks for pointing it out. If this is the prescribed approach I'm fine with keeping it. :thumbsup: